dipy/segment/bundles.py

```diff
@@ -0,0 +1,410 @@
+        model_clust_thr : float
+        reduction_thr : float
+        reduction_distance : string
+            mdf or mam (default mam)
+        slr : bool
+            Use Streamline-based Linear Registration (SLR) locally
+            (default True)
+        slr_metric : BundleMinDistanceMetric
+        slr_x0 : array
+            (default None)
+        slr_bounds : array
+            (default None)
+        slr_select : tuple
+            Select the number of streamlines from model to neirborhood of
+            model to perform the local SLR.
+        slr_method : string
+            Optimization method (default 'L-BFGS-B')
+        pruning_thr : float
+        pruning_distance : string
+            MDF ('mdf') and MAM ('mam')
+
+        Returns
+        -------
+        recognized_bundle : Streamlines
+            Recognized bundle in the space of the original tractogram
+        recognized_labes : array
+            Indices of recognized bundle in the original tractogram
+        recognized_transf : Streamlines
+
+        References
+        ----------
+        .. [Garyfallidis17] Garyfallidis et al. Recognition of white matter
+            bundles using local and global streamline-based registration and
+            clustering.
+        """
+
+        self.reduction_thr = reduction_thr
+        if self.verbose:
+            t = time()
+            print('## Recognize given bundle ## \n')
+
+        self.model_bundle = model_bundle
+        self.cluster_model_bundle(model_clust_thr=model_clust_thr)
+        success = self.reduce_search_space(
+            reduction_thr=reduction_thr,
+            reduction_distance=reduction_distance)
+        if not success:
+            self.pruned_streamlines = None
+            self.transf_streamlines = None
+            self.transf_matrix = None
+            self.labels = []
+            # TODO replace with Streamlines
+            return Streamlines([]), Streamlines([]), Streamlines([])
+        if slr:
+            self.register_neighb_to_model(metric=slr_metric,
+                                          x0=slr_x0,
+                                          bounds=slr_bounds,
+                                          select_model=slr_select[0],
+                                          select_target=slr_select[1],
+                                          method=slr_method)
+        else:
+            self.transf_streamlines = self.neighb_streamlines
+            self.transf_matrix = np.eye(4)
+        self.prune_what_not_in_model(pruning_thr=pruning_thr,
+                                     pruning_distance=pruning_distance)
+
+        if self.verbose:
+            print('Total duration of recognition time is %0.3f sec.\n'
+                  % (time()-t,))
+        # return recognized bundle in original streamlines, labels of
+        # recognized bundle and transformed recognized bundle
+        return self.streamlines[self.labels], self.labels, \
+            self.pruned_streamlines
```

Comment:
It will be good to have detailed documentation about difference between `self.streamlines[self.labels]`  and `self.pruned_streamlines` returned by recognize function.